src/libbson/doc/lifetimes.rst

```diff
@@ -0,0 +1,30 @@
+:man_page: bson_lifetimes
+
+BSON lifetimes
```

Comment:
```suggestion
:symbol:`bson_t` lifetimes
```

Suggest using `BSON` to refer to the data format, and consistently referring to `bson_t` when naming the struct.